### DISCUSSION TITLE
Migration to OAuth 2.0

### DISCUSSION THREAD

**Discussion by deevroman** [streetcomplete/StreetComplete#5404] [Feedback]
I tried the beta version of StreetComplete and there are a few comments.

1. The logout notification was in English in Changelog. As someone who understands the situation, it was clear to me, but will it be clear to ordinary users?

2. Login to the account takes place through a separate browser inside StreetComplete.
- Will password autofill work for everyone in such a browser? I know that on my old phone, autofill is only available in a separate browser
- If you log in via Google (or another social network), you will have to log in to your Google account. If this happened through the browser, the user would have already been logged in. Why do I need another built-in browser?

**Comment by matkoniecz** [comment]
> Why do I need another built-in browser?

separate browser was tried as recommended by Google - it was repeatedly not working in various ways

**Comment by riQQ** [comment]
Context: #5322 / https://github.com/streetcomplete/StreetComplete/releases/tag/v56.0-beta1

**Comment by HolgerJeromin** [comment]
https://github.com/streetcomplete/StreetComplete/issues/413#issuecomment-317567179 for more information about internal versus external browser.

**Reply by westnordost** [reply]
Thanks for that link. I want to add that there is a wide range of other (popular) browsers on Android, which may or may not implement proper callback to the app. Most prominently, the browser bundled with a popular custom rom, I think it was LineageOS, did not do this properly last time I checked.

Relying on an external browser for authorization just seems much too unstable for me, on Android at least. On Desktop it seems to be less of a problem.
